### ISSUE TITLE
[feature] Showing rants only to registered users

### ISSUE THREAD

**Issue by toloveru** [devRant/devRant#361]
In order to help prevent angry project managers, clients and so on from finding particular rants, would it be possible to implement a feature with which we could set a privacy flag while posting that makes the rant only visible when registered and logged in to the platform?

**Comment by DLMousey** [comment]
Surely if the PM/Client/etc is already on devRant looking for it, They'll just sign up if the only thing stopping them is a free account?

If you're worried about someone on the open internet seeing something that you post, Don't post it.

**Comment by toloveru** [comment]
Why would they sign up if they don't know that there's something about them on the website to begin with?
I do fully realize that this won't do much to stop a determined PM, but every little bit helps.

**Comment by Skayo** [comment]
Dislike because embedding a rant or using third party apps (like xmlRant or various terminal clients; even some of my projects)  wouldn't work anymore

**Comment by toloveru** [comment]
Embedding a rant would work as normal if the rant is set to be publicly visible, and only visible to registered users when such privacy settings are enabled, wouldn't it?

**Comment by toloveru** [comment]
To be clear, I don't really mind having all my content on devRant public - this just implies that I do have to watch out a bit with what I say and do, and be lucky that I don't have project managers or clients to worry about yet. It's just that I've seen sometimes on the platform (I particularly remember one rant from azous in the past that apparently generated some issues at work for her) that everything being visible to anyone and their cat (including search engines, data aggregators etc) could sometimes be undesirable.
Actually this is also one of the reasons why I can't really discard Facebook in favor of Twitter or devRant yet. The ability to tailor a post's audience according to the post's content is something that's in my opinion very important to have.
Also, although it would not be as effective, this functionality of making certain content accessible only after login instead of full-blown audience selectors is something that I think is easier to implement. Hence why I mention this as a feature request rather than the latter which I don't think is feasible (yet) but which I would most certainly prefer. Maybe somewhere in the future this could become a thing.

But if the community prefers public posts visible to anyone and their cat, then I will drop my case. It really is saddening but that's democracy.

**Comment by DLMousey** [comment]
If you're posting something that can get you in trouble on any public forum regardless of privacy settings or account-walls you're accepting that risk and the potential consequences of it.

Engage brain before posting requires no effort from Dfox or Trogus and is still significantly more effective than anything that could be implemented for this

**Comment by toloveru** [comment]
While I do agree that basic quality checks are important before posting anything, assuming that anything you post to any audience is really depressing.

Imagine that I make a post visible to only my friends. In that case I trust my friends to not abuse it and respect my desire to only share it with them. If I send a private message to my significant other, I trust them to not share that message with anyone else, for only they were the targeted audience.

But that's advanced audience selectors, something that I think is a rather large move - from having everything public - to make. This feature request on the other hand would allow users to select their desired audience as "only devRant and its users", or "anyone and their cat on the whole wild wild web". The former option would of course also imply that the poster trusts the devRant community to not go talk smack on them or their content to the outside world - it's just the rest that wouldn't get to see it.

**Comment by theseraphim** [comment]
I don't like this feature, though I understand the reasons behind it.

You really shouldn't be posting stuff specific enough to get in trouble, that's a user issue.

Perhaps a warning upon posting or for new users with a link to a guide about successful anonymization of a rant?

I'm happy to write up said guide too.

**Comment by DFoxinator** [comment]
I definitely see your point and the reasoning behind this idea, but as others have pointed out, I think it has a large flaw. I think that a feature like this would provide a false sense of security. A devRant user can be anyone. I feel as if you're not really much more protected with a "devRant users-only" rant as opposed to a public one. And I wouldn't want to make people feel as if they were more protected. Like people have pointed out, it's important to write rants in a way that anonymizes them if you do in fact care about that. 

But I'm curious to hear some more thoughts, I don't want to rule it out, that's just my initial take.

**Comment by toloveru** [comment]
Now that you mention it.. That would indeed be the case. Probably this isn't a good idea after all.
